### DISCUSSION TITLE
Expected performance / load times

### DISCUSSION THREAD

**Discussion by alexsapps** [nextcloud/all-in-one#3655] [Questions] [answered]
What sort of performance could we expect with Nextcloud AIO, assuming a setup which [follows the general recommendations for the best performance](https://github.com/nextcloud/all-in-one/discussions/1335)?

Here are some measurements I'm getting:

* Loading the dashboard with no widgets besides "recent activity":  5 - 15 seconds
* Loading the files app page: ~ 5 seconds
* Loading an individual word document: ~  5 seconds

Some details on my setup:
* I'm using Nextcloud AIO on AWS EC2 (t3.medium + SSD / gp2 volume)
* Only 1 simultaneous user
* SSD volume - I tested with `fio` - the maximum latency is 20 ms, average latency is 3ms, and throughput is 46.1MiB/s
* I can see CPU usage never goes past 50% within a second and usually much lower even while pages are loading
* RAM usage stays between 20-40%

Do the page load times sound right, or what should one expect for a typical installation? How can I know if there is some obvious bottleneck in my control or something easy I can do to improve the performance, or if it's working as expected?

**Comment by szaimen** [comment, marked as answer]
> * gp2 volume

![image](https://github.com/nextcloud/all-in-one/assets/42591237/32b41944-008f-44ed-92d5-880e189cbce5)

> * 20 ms, average latency is 3ms

This doesnt sound like fast storage

**Comment by alexsapps** [comment]
I tried GP3 volume type with 6000 IOPS and I'm not noticing any real difference in page load times. `fio` now says the latency is 24 ms max and 1.3 ms average in the last test.
(Side note, GP2 volume types are burstable to 3000 IOPS, which is what I was getting previously.)

I tried IO2 volume type with only 300 IOPs due to cost and the latency as measured by `fio` got worse. Provisioning 3000 IOPs on IO2 would cost $195/month (for the provisioned IOPs alone) so I haven't tried that.

But I'm surprised that the page load times are barely noticeable between these experiments (always ~5 seconds) and it makes me wonder what the issue is if not storage, CPU or RAM.

How quickly are other folks able to get their dashboard page and files app page to load?

Any opinions on the best AWS setup with reasonable pricing, and what performance can be expected on AWS?

**Comment by DenuxPlays** [comment]
> I tried GP3 volume type with 6000 IOPS and I'm not noticing any real difference in page load times. `fio` now says the latency is 24 ms max and 1.3 ms average in the last test.
> 
> (Side note, GP2 volume types are burstable to 3000 IOPS, which is what I was getting previously.)
> 
> 
> 
> I tried IO2 volume type with only 300 IOPs due to cost and the latency as measured by `fio` got worse. Provisioning 3000 IOPs on IO2 would cost $195/month (for the provisioned IOPs alone) so I haven't tried that.
> 
> 
> 
> But I'm surprised that the page load times are barely noticeable between these experiments (always ~5 seconds) and it makes me wonder what the issue is if not storage, CPU or RAM.
> 
> 
> 
> How quickly are other folks able to get their dashboard page and files app page to load?
> 
> 
> 
> Any opinions on the best AWS setup with reasonable pricing, and what performance can be expected on AWS?

Similar load times...
The nextcloud backend is fast tough but the frontend is very very slow
That's what I noticed

**Reply by alexsapps** [reply]
I saw a few anecdotes online about people achieving 2s load times. I'm not sure how they're doing it and why I can't.

**Reply by DenuxPlays** [reply]
Haha 2s is a dream...
I think that its possible but definitely not without some huge compromises.

**Comment by alexsapps** [comment]
Related discussion:
https://help.nextcloud.com/t/nextcloud-27-all-page-load-times-about-5-seconds-can-it-be-made-faster/171722
